apps/src/userHeaderStatsigReporter/userHeaderStatsigReporter.js

```diff
@@ -1,6 +1,18 @@
 import statsigReporter from '../lib/util/StatsigReporter';
 
-document.addEventListener('DOMContentLoaded', function () {
+if (document.readyState !== 'loading') {
+  console.log('Document is ready');
+  runStatsigReporter();
+} else {
+  document.addEventListener('DOMContentLoaded', function () {
+    console.log('Document is loading');
+    console.log(`Document ready state: ${document.readyState}`);
```

Comment:
Since this is happening in the `DOMContentLoaded` handler, it will presumably always be `"complete"`, which might be interesting to confirm, but doesn't seem crucial.

Reply:
I added this and the one commented on [here](https://github.com/code-dot-org/code-dot-org/pull/60136/files#r1699240205) for extra context as the page is loading. I figured in this case it's better to over-explain than under-explain.